python/cudf/cudf/tests/test_list.py

```diff
@@ -159,3 +159,52 @@ def test_take_invalid(invalid, exception):
     gs = cudf.Series([[0, 1], [2, 3]])
     with exception:
         gs.list.take(invalid)
+
+
+@pytest.mark.parametrize(
+    "data, key, expect",
+    [
+        ([[1, 2, 3], [3, 4, 5], [4, 5, 6]], 4, [False, True, True]),
+        (
+            [[1, 2.0, 3.0], [3.0, 4.0, 5], [4.0, 5, 6.0]],
+            4.0,
+            [False, True, True],
+        ),
+        (
+            [["a", "b", "c"], ["b", "e", "f"], ["e", "g", "m"]],
+            "b",
+            [True, True, False],
+        ),
+        (
+            [[False, True, True], [False, False, False], [True, True, False]],
+            True,
+            [True, False, True],
+        ),
+    ],
+)
+def test_contains(data, key, expect):
+    sr = cudf.Series(data)
+    expect = cudf.Series(expect)
+    got = sr.list.contains(key)
+    assert_eq(expect, got)
+
+
+def test_contains_null_search_key():
+    sr = cudf.Series([[1, 2, 3], [3, 4, 5], [4, 5, 6]])
+    expect = cudf.Series([False, False, False])
+    got = sr.list.contains(key=cudf.Scalar(None))
+    assert_eq(expect, got)
+
+
+def test_contains_null_rows():
+    sr = cudf.Series([[], [], []])
+    expect = cudf.Series([True, True, True])
+    got = sr.list.contains(key=cudf.Scalar(None, dtype="object"))
+    assert_eq(expect, got)
```

Comment:
Interesting, shouldn't this be `[<NA>, <NA>, <NA>]`?

Reply:
@shwina

Reply:
Two thoughts here:

1. We shouldn't accept `None` as the null value here, as it's typically a sentinel hinting at an optional argument. Let's require users to specify `cudf.NA` if they're looking for `NA`:

```python
sr.list.contains(key=cudf.NA, ...)
```

2. Let's return `[False, False, False]`, since in Python `None in []` returns `False` and in Pandas, `pd.NA in pd.Series([])` returns `False`.